contract-hardhat/test/o_load_massive_investor_flow.ts

```diff
@@ -0,0 +1,1175 @@
+            console.log("\nFinal Full Check...");
+            for (const [checkpointIndex, balances] of Object.entries(checkpointBalances)) {
+                const index = parseInt(checkpointIndex);
+                const totalSupply = await stGetter.totalSupplyAt(index);
+                expect(totalSupply).to.equal(totalSupplies[index]);
+                console.log(`\nVerifying Checkpoint ${index}`);
+                console.log(`Expected TotalSupply: ${totalSupplies[index]}, Found: ${totalSupply}`);
+
+                for (const [address, expectedBalance] of Object.entries(balances)) {
+                const actualBalance = await stGetter.balanceOfAt(address, index);
+                expect(actualBalance).to.equal(expectedBalance);
+                }
+            }
+
+            console.log("All checkpoints validated successfully for large investor set");
+        });
+    });
+
+    describe("Check Dividend payouts", async () => {
+        it("Should create a dividend", async () => {
+            const maturity = await latestTime() + 10;
+            const expiry = maturity + duration.days(10);
+            const dividendAmount = ethers.parseEther("10000");
+
+            //Make sure token_owner has enough POLY tokens
+            await I_PolyToken.connect(token_owner).getTokens(dividendAmount, token_owner.address);
+
+            //Approve spending
+            await I_PolyToken.connect(token_owner).approve(I_ERC20DividendCheckpoint.target, dividendAmount);
+
+            //Create dividend
+            const tx = await I_ERC20DividendCheckpoint.connect(token_owner).createDividend(
+                maturity,
+                expiry,
+                I_PolyToken.target,
+                dividendAmount,
+                dividendName
+            );
+
+            const receipt = await tx.wait();
+            const event = receipt.logs
+                .map(log => {
+                try {
+                    return I_ERC20DividendCheckpoint.interface.parseLog(log);
+                } catch {
+                    return null;
+                }
+                })
+                .find(e => e && e.name === "ERC20DividendDeposited");
+
+            expect(event).to.not.be.null;
+            const dividendIndex = event!.args._dividendIndex;
+
+            // Check internal dividend data
+            const data = await I_ERC20DividendCheckpoint.getDividendsData();
+            expect(data[1][Number(dividendIndex)]).to.equal(BigInt(maturity));
+            expect(data[2][Number(dividendIndex)]).to.equal(BigInt(expiry));
+            expect(data[3][Number(dividendIndex)]).to.equal(dividendAmount);
+            expect(data[4][Number(dividendIndex)]).to.equal(0n); // nothing claimed yet
+            expect(data[5][Number(dividendIndex)]).to.equal(dividendName);
+
+            console.log(`Dividend #${dividendIndex} created and verified`);
+        });
+
+        it("Investors should be able to pull their own dividend", async () => {
```

Comment:
Can create a test to calculate the dividend they are supposed to get and make sure it matches what they get ?